sagemaker/sagemaker.w

```diff
@@ -0,0 +1,37 @@
+bring aws;
+bring util;
+bring "./types.w" as types;
+bring "./sagemaker.sim.w" as s;
+bring "./sagemaker.tfaws.w" as t;
+
+pub class Endpoint impl types.ISageMaker {
+  pub endpointName: str;
+
+
+  inner: types.ISageMaker;
+
+  new(endpointName: str, inferenceComponentName: str) {
+    this.endpointName = endpointName;
+
+
+    let target = util.env("WING_TARGET");
+    if target == "sim" {
+      if std.Node.of(this).app.isTestEnvironment {
+        // in case of test running on sim, use simulator version
+        this.inner = new s.SageMaker_sim(endpointName, inferenceComponentName) as "sim"; 
+      } else {
+        // in case of running on sim interactively (in development mode), use AWS version
+        this.inner = new t.SageMaker_tfaws(endpointName, inferenceComponentName) as "tf-aws";  
```

Comment:
```suggestion
        this.inner = new t.SageMaker_tfaws(util.env("SAGEMAKER_ENDPOINT_NAME"), inferenceComponentName) as "tf-aws";  
```

Maybe the right DX is to allow this to be defined through environment variables when developing interactively.

Reply:
Good point about multiple instances, but I think it's still useful to support somehow specifying these externally to the app because then users would need a special case for the simulator

Reply:
I agree, but I think it's part of a bigger "problem" that is shared across a few libraries with a similar structure (which I copied from bedrock), I think I'll open an issue for that :)

Reply:
https://github.com/winglang/wing/issues/5629

Reply:
Maybe this is something we can support through `wing.yaml` that @hasanaburayyan is working on

Reply:
👋 Howdy, yea https://github.com/winglang/wing/pull/5557 introduces some features that could potentially help out here. 

For instance there is a `parameterRegistrar` that exists on the root App, that can be used to register/read parameters. Though, its currently marked as internal (which maybe it shouldnt be)

so for instance you could register this parameter with something like: 
`std.Node.of(this).app.parameterRegistrar.addParameterSchema(some_json_schema)`

The validation of the parameters happens before the construct tree is synthesized so, we can tell users exactly what they are missing. Also potentially in another iteration add tools for providing these parameters interactively.

Reply:
and we can also have a default value, right? I'll add another issue to change it here and on bedrock as soon as it is merged :)

Reply:
right now we can workaround it by:
```
let var endpointName = util.env("ENDPOINT_NAME"); // prod one
if (!app.isTestEnvironment &&  util.env("WING_TARGET") == "sim") {
  endpointName = <some value read from some file>; // sim non-test one
}
```
if needed

Reply:
> and we can also have a default value

Technically JSON schema supports explicitly stating defaults, however they have no real effect. So even if the schema defined a default it would still be up to your code to enforce it. (its essentially just docs)